components/eamxx/src/share/io/scream_io_utils.cpp

```diff
@@ -52,18 +66,29 @@ std::string find_filename_in_rpointer (
   if (not found) {
     broadcast_string(content,comm,comm.root_rank());
 
-    // If the history restart file is not found, it must be because the last
-    // model restart step coincided with a model output step, in which case
-    // a restart history file is not written.
-    // If that's the case, *disable* output restart, by setting
-    //   'Restart'->'Perform Restart' = false
-    // in the input parameter list
-    EKAT_ERROR_MSG (
-        "Error! Restart requested, but no restart file found in 'rpointer.atm'.\n"
-        "   restart filename prefix: " + filename_prefix + "\n"
-        "   restart file type: " + std::string(model_restart ? "model restart" : "history restart") + "\n"
-        "   run t0           : " + run_t0.to_string() + "\n"
-        "   rpointer content:\n" + content);
+    if (model_restart) {
+      EKAT_ERROR_MSG (
+          "Error! Restart requested, but no model restart file found in 'rpointer.atm'.\n"
+          "   model restart filename prefix: " + filename_prefix + "\n"
+          "   model restart filename pattern: " + pattern_str + "\n"
+          "   run t0           : " + run_t0.to_string() + "\n"
+          "   rpointer content:\n" + content + "\n\n");
+    } else {
+      EKAT_ERROR_MSG (
+          "Error! Restart requested, but no history restart file found in 'rpointer.atm'.\n"
+          "   hist restart filename prefix: " + filename_prefix + "\n"
+          "   hist restart filename pattern: " + pattern_str + "\n"
+          "   run t0           : " + run_t0.to_string() + "\n"
+          "   avg_type         : " + e2str(avg_type) + "\n"
+          "   output freq      : " + std::to_string(control.frequency) + "\n"
+          "   output freq units: " + control.frequency_units + "\n"
+          "   rpointer content:\n" + content + "\n\n"
+          " Did you change output specs (avg type, freq, or freq units) across restart? If so, please, remember that it is not allowed.\n"
+          " It is also possible you are using a rhist file create before commit 6b7d441330d. That commit changed how rhist file names\n"
+          " are formed. In particular, we no longer use INSTANT.${REST_OPTION}_x${REST_N}, but we use the avg type, and freq/freq_option\n"
+          " of the output stream (to avoid name clashes if 2 streams only differ for one of those). If you want to use your rhist file,\n"
+          " please rename it, so that the avg-type, freq, and freq_option reflect those of the output stream.\n");
+    }
```

Comment:
Well, a restart is a restart. The if-else is to print different messages, since model/history restarts have different specs. Also, the message for hist restart contains the warning about the bwd incompatible change, which includes a comment stating how one can change an old rhist file name so that it works for the new version.